src/main/java/gameframework/drawing/DrawableImage.java

```diff
@@ -25,19 +27,19 @@ public DrawableImage(URL imageUrl, GameCanvas gameCanvas) {
 	 * @param imageUrl
 	 *            is the path to the image
 	 */
-	protected void handleImage(URL imageUrl) {
+	protected void handleImage(URL imageUrl) throws IllegalArgumentException {
 		Toolkit toolkit = Toolkit.getDefaultToolkit();
 		image = toolkit.createImage(imageUrl);
 		MediaTracker tracker = canvas.createMediaTracker();
 		tracker.addImage(image, 0);
+		
 		try {
 			tracker.waitForAll();
 			if (tracker.isErrorAny()) {
-				throw new IllegalArgumentException(
-						"Problem while loading an image " + imageUrl);
+				throw new IllegalArgumentException("Problem while loading an image " + imageUrl);
 			}
 		} catch (InterruptedException e) {
-			throw new RuntimeException(e);
+			tracker.removeImage(image);
```

Comment:
Rather than throw an runtime exception and eventually cause the stop of the game, I prefer to remove the image to improve stability.